packages/mds-ingest-service/repository/mappers/event-mappers.ts

```diff
@@ -50,6 +53,30 @@ export const EventEntityToDomain = ModelMapper<EventEntityModel, EventDomainMode
   }
 )
 
+export type PartialEventEntityModel = Omit<DeepPartial<EventEntityModel>, 'telemetry'> & {
+  telemetry: Partial<TelemetryEntityModel>
+}
+
+export const PartialEventEntityToDomain = ModelMapper<PartialEventEntityModel, PartialEventDomainModel, {}>(
+  (entity, options) => {
+    const {
+      id,
+      telemetry: telemetry_entity,
+      annotation: annotation_entity,
+      migrated_from_source,
+      migrated_from_version,
+      migrated_from_id,
+      ...domain
+    } = entity
+
+    const telemetry = telemetry_entity as DeepPartial<TelemetryDomainModel>
+    const annotation: Nullable<EventAnnotationDomainModel> = annotation_entity
+      ? EventAnnotationEntityToDomain.map(annotation_entity as EventAnnotationEntity)
```

Comment:
We definitely shouldn't have to do all of these `as` casts. Make sure that your `PartialEventEntityModel` is correct